### PR TITLE
Features/233 matic bsc

### PR DESCRIPTION
**Pull request by ernestpascual**
## Description of Changes
Support for BSC and Matic chains. Changed BPT to SLP and added deployed contracts in BSC and MATIC.

[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDF233-)

### How to test
For BSC Checklist:
- You must have SLP on either of the pools
- You must have HALO to test vesting
- env file is on 1password (halodao-interface-test.env)

#### BSC Scan to trigger contract functions
Rewards Manager: https://testnet.bscscan.com/address/0x057295be0A59E97F8d87c4d045590a068342e8d5#code
Rewards: https://testnet.bscscan.com/address/0x8edDA0534e1D12FfE24d6aee13BF8Fb820C46271#code

How to test:
- Test vesting
- Test if the pools are loading (only bsc for now, you can get SLP tokens by clicking Manage then clicking "to stake..." link

Note:
- you can release epoch from the bscscan using the deployer wallet
- you can trigger rewards contract functions using the deployer wallet

### Developer Checklist:

* [x] I have followed the guidelines in our Contributing document
* [x] This PR has a corresponding JIRA ticket
* [x] My branch conforms with our naming convention i.e. `feature/HDF-XXX-description`
* [x] All files have appropriate file headers and documentation
* [x] I have written new tests for your core changes, as applicable
* [x] I have successfully ran tests locally

### Reviewers Checklist:
* [ ] Code is readable and understandable; any unclear parts have explanations 
* [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
* [ ] I have successfully ran tests locally

### CODE REVIEW

File: src/constants/index.ts [hunk 1]
```diff
@@ -66,29 +66,36 @@ export const HALO: ChainTokenMap = {
...
     ChainId.BSC_TESTNET,
     process.env.REACT_APP_HALO_TOKEN_ADDRESS_BSC_TESTNET || '',
     18,
-    'HALO',
-    'HALOToken'
+    'RNBW',
+    'RNBWToken'
+  ),
+  [ChainId.MATIC_TESTNET]: new Token(
+    ChainId.BSC_TESTNET,
```
Comment (schystz) [inline src/constants/index.ts]:
```suggestion
    ChainId.MATIC_TESTNET,
```

**Review by schystz — COMMENTED** [review]
1 comment for now. More review soon when the other branches merged to this branch has been merged to `develop`. Quite difficult right now to identify which changes are introduced by this branch only.

File: src/constants/index.ts (hunk 1 above)
Comment (ernestpascual) [reply src/constants/index.ts]: thanks for spotting!

**Comment by schystz** [comment]
> 1 comment for now. More review soon when the other branches merged to this branch has been merged to `develop`. Quite difficult right now to identify which changes are introduced by this branch only.

Actually @ernestpascual , can you change the base branch from master to develop? That's probably the reason why we are seeing too many changes :)

Would be good if you can sync the branch with latest changes on develop as well.

File: public/locales/en.json [hunk 2]
```diff
@@ -103,7 +103,9 @@
   "valueStaked": "Value Staked",
   "earned": "Earned",
   "getBPTTokens": "To stake, get BPT Tokens here",
+  "getSLPTokens": "To stake, get SLP Tokens here",
   "closeTxt": "Close",
+  "stakeToEarnSushi": "Stake your Sushi (Liquidty Pool) Pool Tokens (SLP) to earn Halo!",
```
Comment (schystz) [inline public/locales/en.json]:
```suggestion
  "stakeToEarnSushi": "Stake your Sushi (Liquidty Pool) Pool Tokens (SLP) to earn Rainbow Candy (RNBW)!",
```

File: package.json [hunk 3]
```diff
@@ -131,6 +131,7 @@
   },
   "license": "GPL-3.0-or-later",
   "dependencies": {
+    "@sushiswap/sushi-data": "^1.0.26",
```
Comment (schystz) [inline package.json]: Where is this package used @ernestpascual ?

File: src/constants/index.ts [hunk 4]
```diff
@@ -97,25 +104,31 @@ export const HALO_TOKEN_ADDRESS: ChainAddressMap = {
...
   [ChainId.KOVAN]: process.env.REACT_APP_HALOHALO_ADDRESS_KOVAN || '',
   [ChainId.RINKEBY]: process.env.REACT_APP_HALOHALO_ADDRESS_RINKEBY || '',
   [ChainId.GÖRLI]: process.env.REACT_APP_HALOHALO_ADDRESS_GOERLI || '',
-  [ChainId.BSC_TESTNET]: process.env.REACT_APP_HALOHALO_ADDRESS_BSC_TESTNET || ''
+  [ChainId.BSC_TESTNET]: process.env.REACT_APP_HALOHALO_ADDRESS_BSC_TESTNET || '',
+  [ChainId.MATIC_TESTNET]: process.env.REACT_APP_HALOHALO_ADDRESS_MATIC_MUMBAI || ''
 }
 
+// Sushi Url
+export const SUSHI_SUBGRAPH_URL = process.env.REACT_APP_SUSHI_SUBGRAPH_URL || ''
```
Comment (schystz) [inline src/constants/index.ts]: This one too @ernestpascual . Seems unused atm?

**Review by schystz — COMMENTED** [review]
@ernestpascual site is loading very slowly for me locally, seeing this from the console log and counting -
Screen Shot 2021-06-16 at 11 49 19 AM

Please see also a few small things below.

**Comment by schystz** [comment]
> @ernestpascual site is loading very slowly for me locally, seeing this from the console log and counting -
Screen Shot 2021-06-16 at 11 49 19 AM

Looks like this is only happening if I used an unsupported network (not Kovan or BSC Testnet)

File: package.json (hunk 3 above)
Comment (ernestpascual) [reply package.json]: removed this, was just testing stuff !

File: src/constants/index.ts (hunk 4 above)
Comment (ernestpascual) [reply src/constants/index.ts]: yup, cleaning up!

**Comment by ernestpascual** [comment]
@schystz  @barryhalo  i have merge issues, let's schedule a call tomorrow :)

**Review by schystz — APPROVED** [review]
Looks good after the review feedback! Only thing left to do is to cater main networks (ethereum, bsc & polygon) once we deploy the contracts next week.

**Review by tracyarciaga — APPROVED** [review]
Looks good. Local test working too.

**Pull request merged**